lib/bundler_api/dep_fetcher.rb

```diff
@@ -0,0 +1,167 @@
+require 'cgi'
+require 'open-uri'
+require 'set'
+require 'bundler_api'
+require 'bundler_api/dep_calc'
+require 'bundler_api/gem_helper'
+require 'bundler_api/metriks'
+require 'bundler_api/update/job'
+
+class BundlerApi::DepFetcher
+  def initialize(memcached_client)
+    @memcached_client = memcached_client
+    @fetchers = []
+  end
+
+  def <<(fetcher)
+    @fetchers << fetcher
+  end
+
+  def fetch(gems)
+    results = Results.new(@memcached_client, gems)
+
+    @fetchers.each do |fetcher|
+      break if results.done?
+      fetcher.fetch(results)
+    end
+
+    results.remaining_keys.each do |key|
+      @memcached_client.set(key, [])
+    end
```

Comment:
You already have all the remaining keys and the memcached client in the results. Does it make any sense to push this behavior in there?